src/audio/rtnr/rtnr.c

```diff
@@ -494,62 +498,33 @@ static int rtnr_reconfigure(struct comp_dev *dev)
 	return 0;
 }
 
-static int rtnr_set_comp_config_by_ipc_config_process(struct comp_dev *dev,
-							struct ipc_config_process *config)
-{
-	struct comp_data *cd = comp_get_drvdata(dev);
-	int ret;
-
-	comp_dbg(dev, "rtnr_set_comp_config_by_ipc_config(): %d", config->size);
-	if (config->size != sizeof(cd->config)) {
-		comp_err(dev, "rtnr_set_comp_config_by_ipc_config(): invalid size %d",
-			config->size);
-		return -EINVAL;
-	}
-
-	ret = memcpy_s(&cd->config,
-				sizeof(cd->config),
-				config->data,
-				config->size);
-	if (ret)
-		return ret;
-
-	comp_info(dev,
-		"rtnr_set_comp_config_by_ipc_config(): sample_rate = %d, enabled=%d",
-		cd->config.params.sample_rate,
-		cd->config.params.enabled);
-
-	return ret;
-}
-
-static int rtnr_set_comp_config_by_ipc_ctrl_data(struct comp_dev *dev,
-							struct sof_ipc_ctrl_data *cdata)
+static int rtnr_set_config_bytes(struct comp_dev *dev,
+				 unsigned char *data, uint32_t size)
 {
 	struct comp_data *cd = comp_get_drvdata(dev);
 	int ret;
 
-	comp_dbg(dev, "rtnr_set_comp_config_by_ipc_ctrl_data(): size: %d",
-		cdata->data->size);
-
-	if (cdata->data->size != sizeof(cd->config)) {
-		comp_err(dev,
-			"rtnr_set_comp_config_by_ipc_ctrl_data(): invalid size %d",
-			cdata->data->size);
-
+	/*
+	 * The received data could be the combined blob of the control
+	 * widgets defined in the topology, or the config received by
+	 * SOF_CTRL_CMD_BINARY. In either case we just have to check if
+	 * the whole config data is received.
+	 */
+	if (size < sizeof(cd->config)) {
+		comp_err(dev, "rtnr_set_config_data(): invalid size %d",
+			 size);
 		return -EINVAL;
 	}
 
 	ret = memcpy_s(&cd->config,
-				sizeof(cd->config),
-				cdata->data->data,
-				cdata->data->size);
-	if (ret)
-		return ret;
+		       sizeof(cd->config),
+		       data,
+		       sizeof(cd->config));
 
 	comp_info(dev,
-		"rtnr_set_comp_config_by_ipc_ctrl_data(): sample_rate = %d, enabled=%d",
-		cd->config.params.sample_rate,
-		cd->config.params.enabled);
+		  "rtnr_set_config_data(): sample_rate = %d, enabled=%d",
```

Comment:
sample_rate = %u

Reply:
Since this a cherry-pick PR and the original commit was merged into main and adl-004-stable branches, I've created another PR https://github.com/thesofproject/sof/pull/6566 to fix the formatting issue. Thanks.

Reply:
Got it. Thanks

Reply:
Hi @yaochunhung https://github.com/thesofproject/sof/pull/6566 has been merged into main. Please check&merge this PR so that I could cherry-pick the patch. Thank you!

Reply:
@MingJenTai  done, thanks.